docs/examples/ssh/SSH_BITBUCKET.md

```diff
@@ -0,0 +1,61 @@
+# SSH service in Bitbucket on Kubernetes
+
+In addition to providing a service on HTTP(S), Bitbucket also allows remote Git operations over SSH connections. By default, Kubernetes Ingress controllers only work for HTTP connections, but some ingress controllers also support TCP connections.
+
+Depending on the need of your deployment, SSH access can be provided through two mechanisms:
+1. Opening the TCP port through the ingress controller - This option should be used if the SSH service is required to be available on the same DNS name as the HTTP service.
+2. Creating a separate Kubernetes `LoadBalancer` service - This option is available if the ingress controller does not support TCP connections, or if you don’t need your deployment to have the SSH service available on the same DNS name as the HTTP service.
+
+## Setting up the NGINX Ingress Controller to accept SSH connections
```

Comment:
I think we have to scrap this option for now. Dylan just tested it again on Kubernetes 1.19 and it does not seem to work as expected. In my testing I ran into a bug that I thought was unrelated to I didn't think much of it, but the combination of using HTTPS with the proxy protocol as well as a TCP service on the same kubernetes service (and therefore ELB on AWS) does not currently work. There's the possibility that it will work on Kubernetes 1.20 with the `appProtocol` field, but this is not tested yet.

Reply:
This works for me on an on-prem k8s cluster. It will even work on Minikube. Why does k8s matter here if it's just all about port forwarding from Nginx to service directly?

Reply:
So, here's the configmap:

```
eugene:~$ kubectl get cm/tcp-services -n ingress-nginx -o yaml
apiVersion: v1
data:
  "7999": git/bitbucket:7999
kind: ConfigMap
  name: tcp-services
  namespace: ingress-nginx

```
And git clone works just fine:

```
eugene:/tmp$ git clone ssh://git@git.k8sdemo.tk:7999/k8s/kubernetes.git
Cloning into 'kubernetes'...
remote: Enumerating objects: 32, done.
remote: Counting objects: 100% (32/32), done.
remote: Compressing objects: 100% (16/16), done.
remote: Total 32 (delta 3), reused 0 (delta 0), pack-reused 0
Receiving objects: 100% (32/32), done.
Resolving deltas: 100% (3/3), done.
```
Since this is my on-prem self made k8s cluster, Nginx listens on host port on the node (host network=true) but it's just a matter of configuration of course. In clouds Nginx service will be LoadBalancer in most cases.

My cluster is v1.18.10

Reply:
@bianchi2 it would be good to understand how you have configured your `ingress-nginx` controller. We're basing the installation and config on our ingress readme (still in review) here: https://github.com/atlassian-labs/data-center-helm-charts/blob/dckube-338-eks-setup-examples/docs/examples/ingress/INGRESS_NGINX.md#2-create-a-custom-config-yaml

Reply:
@errcode1202 like I said, my cluster is an on-prem self made one. Nginx is deployed with hostNetwork=true. As per docs, you need to add this additional ssh port to Nginx service. I haven't tried it in EKS, though it did work for me on Digital Ocean and self made K8s clusters with Nginx deployed (hostNetwork). I'll give it a go myaself on EKS and open a PR if my tests are successful.

Reply:
Yep, if it's on-prem it should work, however, we are giving examples for AWS and with that it won't work. If we specifically call out that it won't work with AWS (and potentially other cloud providers) then we might be able to keep this section.

Reply:
Okay - In the meantime Im testing the alternative approach to see if thats a viable fallback for AWS